api/graphql/types/recurring_reservation/serializers.py

```diff
@@ -38,52 +47,372 @@ class Meta:
+        model = Reservation
+        fields = [
+            #
+            "name",
+            "description",
+            "num_persons",
+            "state",
+            "type",
+            "working_memo",
+            #
+            "buffer_time_before",
+            "buffer_time_after",
+            "handled_at",
+            "confirmed_at",
+            #
+            "applying_for_free_of_charge",
+            "free_of_charge_reason",
+            #
+            "reservee_id",
+            "reservee_first_name",
+            "reservee_last_name",
+            "reservee_email",
+            "reservee_phone",
+            "reservee_organisation_name",
+            "reservee_address_street",
+            "reservee_address_city",
+            "reservee_address_zip",
+            "reservee_is_unregistered_association",
+            "reservee_language",
+            "reservee_type",
+            #
+            "billing_first_name",
+            "billing_last_name",
+            "billing_email",
+            "billing_phone",
+            "billing_address_street",
+            "billing_address_city",
+            "billing_address_zip",
+            #
+            "user",
+            "purpose",
+            "home_city",
+        ]
+        extra_kwargs = {field: {"required": field in ["type", "user"]} for field in fields}
+
+
+class ReservationSeriesSerializer(RecurringReservationCreateSerializer):
+    """Create the recurring reservation with all its reservations."""
+
+    reservation_details = ReservationSeriesReservationSerializer(
+        required=True,
+        write_only=True,
+        validators=[input_only_field],
+    )
+    check_opening_hours = serializers.BooleanField(
+        required=False,
+        write_only=True,
+        validators=[input_only_field],
+    )
+    skip_dates = serializers.ListField(
+        child=serializers.DateField(),
+        required=False,
+        allow_empty=True,
+        write_only=True,
+        validators=[input_only_field],
+    )
+
+    class Meta(RecurringReservationCreateSerializer.Meta):
+        fields = [
+            *RecurringReservationCreateSerializer.Meta.fields,
+            "reservation_details",
+            "check_opening_hours",
+            "skip_dates",
+        ]
+
+    def save(self, **kwargs: Any) -> RecurringReservation:
+        reservation_details: dict[str, Any] = self.initial_data.get("reservation_details")
+        skip = self.initial_data.get("skip_dates", [])
+        check_opening_hours = self.initial_data.get("check_opening_hours", False)
+
+        # Create both the recurring reservation and the reservations in a transaction.
+        # This way if we get, e.g., overlapping reservations, the whole operation is rolled back.
+        with transaction.atomic():
+            instance = super().save()
+            self.create_reservations(instance, reservation_details, skip, check_opening_hours)
+
+        return instance
+
+    def create_reservations(
```

Comment:
This method is missing some required validation checks for when to allow creating reservations.
For example, see:
- [ReservationUnitFirstReservableTimeHelper._get_hard_closed_time_spans](https://github.com/City-of-Helsinki/tilavarauspalvelu-core/blob/main/reservation_units/utils/first_reservable_time_helper.py#L377)
- [ReservationUnitFirstReservableTimeHelper._get_soft_closed_time_spans](https://github.com/City-of-Helsinki/tilavarauspalvelu-core/blob/main/reservation_units/utils/first_reservable_time_helper.py#L429)

Reply:
I think since recurring reservations follow staff reservation creation rules, we don't need to validate these ones. @vergama?

Reply:
Yes, they do not limit staff reservations.